### PR TITLE
TSQL - Implement unary and binary expressions with correct precedence

### PR DESCRIPTION
**Pull request by jimidle**
Unary operators were not correctly implemented in the ANTLR grammar file, and this is now corrected. In addition, unary operators are right associative, which affects order of operations in certain case. Precedence has also been corrected to follow the TSQL normative specification. 

Progresses: #275

### CODE REVIEW

**Comment by jimidle** [comment]
@vil1 @nfx I have another PR based on this one that I would like to submit if eiether of you have time to check and merge this.

File: core/pom.xml [hunk 1]
```diff
@@ -87,6 +87,12 @@
       <version>3.3.0.0-alpha.1</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>4.1.0</version>
```
Comment (vil1) [inline core/pom.xml:93]:
```suggestion
      <version>${mockito.version}</version>
```

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/tsql/TSqlLexer.g4 [hunk 2]
```diff
@@ -1212,6 +1212,8 @@ IPV4_ADDR: DEC_DIGIT+ '.' DEC_DIGIT+ '.' DEC_DIGIT+ '.' DEC_DIGIT+;
 SPACE: [ \t\r\n]+ -> skip;
 // https://docs.microsoft.com/en-us/sql/t-sql/language-elements/slash-star-comment-transact-sql
 COMMENT      : '/*' (COMMENT | .)*? '*/' -> channel(HIDDEN);
+//TODO: In ANTLR v4 we cannot distinguish between 7 + ---42 amd  -- A comment. Needs custom char stream
```
Comment (vil1) [inline core/src/main/antlr4/com/databricks/labs/remorph/parsers/tsql/TSqlLexer.g4:1215]:
```suggestion
//TODO: In ANTLR v4 we cannot distinguish between 7 + ---42 and  -- A comment. Needs custom char stream
```

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala [hunk 3]
```diff
@@ -54,14 +54,24 @@ case class UnresolvedOperator(unparsed_target: String) extends Expression {}
...
+// Binary Arithmetic expressions
 case class Multiply(left: Expression, right: Expression) extends Binary(left, right) {}
 case class Divide(left: Expression, right: Expression) extends Binary(left, right) {}
 case class Mod(left: Expression, right: Expression) extends Binary(left, right) {}
+
 case class Add(left: Expression, right: Expression) extends Binary(left, right) {}
 case class Subtract(left: Expression, right: Expression) extends Binary(left, right) {}
 
-// Binary bitwise expressions
+// Binary bitwise expressions and con
```
Comment (vil1) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala:74]:
```suggestion
// Binary bitwise expressions
```

**Review by vil1 — APPROVED** [review]
LGTM (modulo a few minute, non-blocking suggestions)

File: core/pom.xml (hunk 1 above)
Comment (jimidle) [reply core/pom.xml:93]: Mockito was added by someone else. It needs to be in the Pom or IntelliJ will not run any tests because snowflake uses it.

**Comment by jimidle** [comment]
Please merge so I can rebase the other outstanding PRs :)

**Comment by jimidle** [comment]
> LGTM (modulo a few minute, non-blocking suggestions)

@vil1 Do you have merge permissions on this repo? I am not completely blocked, but the other PRs will need to remain WIP until this one is merged. 

#307-> #308  -> #309

**Comment by vil1** [comment]
Sorry but I don't have the required permissions

**Comment by sundarshankar89** [comment]
@nfx  Need your attention codecov has been waiting on status for last 6+ hours.

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Pull request merged**